### PR TITLE
[DS-2779] Fix versioning due to service api

### PR DESCRIPTION
**Pull request by KevinVdV**
This pull request has one controversial change in it, it makes the DSpaceObject, Handle relation a OneToMany.

If versioning is enabled a single item can have multiple handles. I have kept the getHandle() method on DSpaceObject for now since it has over 600 usages.
The getHandle() will always attempt to return the non versioned handle, if ever required a getHandles() method is also available.

### CODE REVIEW

**Comment by tdonohue** [comment]
This breaks several unit tests in `SupervisedItemTest` (see Travis CI for more info). At a minimum those would need to be fixed (and this change might warrant additional unit tests).

**Comment by KevinVdV** [comment]
Added an additional commit containing additional unit tests for the handles. It brought to light some things that weren't working as expected, so it is great to have them.
I haven't been able to fix the unit tests just yet (I don't understand why they are breaking ? The handle fix is unrelated to workspace items....)

**Comment by pnbecker** [comment]
Before we decide about this PR I kindly ask to discuss my PRs against DS-1348 and the ticket itself. I see some problems with the concept of "non versioned handles". For more details please take a look in DS-1348 and its comments.

**Comment by pnbecker** [comment]
For backward compatability we do not have a choice here. If anyone used the versioning feature as implemented since DSpace 4 we will need this patch.

I rebased this PR and added some more changes: The config/spring/api/versioning-service.xml had a minor error and the JSPUI had problems deleting Versions. I created a PR with those changes against Kevin's branch. In case it is more handy to merge this one first and create another PR then, I can do that as well.

**Comment by KevinVdV** [comment]
I was able to fix the unit tests. I'm not quite pleased with the solution as it requires a context.complete() to trigger the sql that links the groups to the workspace items.
I have no idea how my versioning fixes influenced this behavior.

It shouldn't cause any issues in the code since there will be little need to first add & then retrieve by query .

**Comment by pnbecker** [comment]
Thanks @KevinVdV for taking care of this. Would be great if we could merge this asap!

At the DINI-Jahrestagung (conference in the last two days) I talked to someone who resolved DS-1349 for XMLUI. I would like to ask him to create a PR, but this would be much simpler if this PR gets merged first.

+1

File: dspace-jspui/src/main/java/org/dspace/app/webui/components/VersioningItemHome.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public void process(Context context, HttpServletRequest request,
 
 			if (latestVersion != null) {
 				if (latestVersion != null && latestVersion.getItem() != null
```
Comment (mwoodiupui) [inline dspace-jspui/src/main/java/org/dspace/app/webui/components/VersioningItemHome.java]: The first test here is redundant.

Comment (pnbecker) [inline dspace-jspui/src/main/java/org/dspace/app/webui/components/VersioningItemHome.java]: Yes it is. Will be removed by #900

**Comment by peterdietz** [comment]
+1 We aren't using versioning anywhere, but if people are satisfied with this, then it sounds good to me.

**Comment by mwoodiupui** [comment]
+1 by inspection.  We aren't using versioning yet, but what I see looks sound.

**Pull request merged**